src/modules/sub-items/components/table-view/table.view.columns.toggler.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+
+import Icon from '../../../common/icon/icon';
+import TableViewColumnsTogglerListElement from './table.view.columns.toggler.list.element';
+import { headerLabels } from './table.view.component';
+
+export default class TableViewColumnsTogglerComponent extends Component {
+    constructor(props) {
+        super(props);
+
+        this.togglePanel = this.togglePanel.bind(this);
+        this.hidePanel = this.hidePanel.bind(this);
+
+        this._refTogglerButton = createRef();
+
+        this.state = {
+            isOpen: false,
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('click', this.hidePanel, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('click', this.hidePanel);
+    }
+
+    hidePanel({ target }) {
+        if (!this.state.isOpen) {
+            return;
+        }
+
+        const isClickInsidePanel = target.closest('.c-table-view-columns-toggler__panel');
+        const isClickOnBtn = target.closest('.c-table-view-columns-toggler__btn');
```

Comment:
nitpick: both are inside `c-table-view-columns-toggler` you can do closest only once.